src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -337,6 +338,24 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
         }
 
         /// <summary>
+        /// Try to get the directory this assembly is in. Returns null if assembly
+        /// was in the GAC.
+        /// </summary>
+        private static string TryGetClientDir()
+        {
+            var assembly = typeof(BuildClient).Assembly;
```

Comment:
It probably doesn't matter a whole lot, but the ```typeof(BuildClient).Assembly``` was chosen based on the code's original location. Perhaps it should be ```typeof(ManagedCompiler).Assembly``` now.